Expose the private token and storefrontId within Hydrogen config (#1998)
Shopify/hydrogen: .changeset/empty-rocks-lick.md

---
'@shopify/hydrogen': patch
'create-hydrogen-app': patch
---

> We've exposed the private server-to-server storefront API token in the hydrogen config. The private token is necessary when deploying to production, otherwise the requests to the storefront API will be rate limited. This should make it easier to properly configure hydrogen when deploying to non-Oxygen environments. We'll also warn when in production mode and the token is not defined.

Reviewer: ```suggestion
We've exposed the private server-to-server Storefront API token in the Hydrogen config. The private token is necessary when deploying to production, otherwise the requests to the storefront API will be rate limited. This should make it easier to properly configure Hydrogen when deploying to non-Oxygen environments. We'll also warn when in production mode and the token is not defined.
```

> We've exposed the private server-to-server storefront API token in the hydrogen config. The private token is necessary when deploying to production, otherwise the requests to the storefront API will be rate limited. This should make it easier to properly configure hydrogen when deploying to non-Oxygen environments. We'll also warn when in production mode and the token is not defined.

Reviewer 2: Copy edits!

```suggestion
We've exposed the private server-to-server Storefront API token in the Hydrogen config file. This private token is required when deploying to production, otherwise the requests to the storefront API will be rate-limited. This change will make it easier to configure Hydrogen when deploying to non-Oxygen environments. We'll also display a warning in production mode if this token is not defined.
```

> We've also added the `storefrontId` property to the config. It's necessary to add this as well, otherwise your analytics dashboard on the admin will be broken.

Reviewer: ```suggestion
We've also added the `storefrontId` property to the config. You need to add this, otherwise your analytics dashboard in the Shopify admin will break.
```

> We've also added the `storefrontId` property to the config. It's necessary to add this as well, otherwise your analytics dashboard on the admin will be broken.

Reviewer 2: ```suggestion
We've also added the `storefrontId` property to the config. This enables Hydrogen data to display properly in the Shopify admin analytics dashboard.
```

> Lastly we also updagted all Oxygen environment variable references to a more consistent naming convention. The previous variables are still available, but are deprecated, and will be removed down the road. If you still reference them, a warning will be given:

Reviewer: ```suggestion
Last, we also updated all Oxygen environment variable references to a more consistent naming convention. The previous variables are still available, but are deprecated, and will be removed down the road. If you still reference them, a warning will be given.

The following are the new variable names:
```

> Lastly we also updagted all Oxygen environment variable references to a more consistent naming convention. The previous variables are still available, but are deprecated, and will be removed down the road. If you still reference them, a warning will be given:

Reviewer 2: ```suggestion
Lastly, we've updated all Oxygen environment variables to a more consistent naming convention. The previous variables are still available, but are deprecated, and will be removed in the future. You’ll see a warning in your console if you use the old environment variables. You can update your variable references using this table:
```

> | **Current Oxygen Variable**         | **New Oxygen Variable**     |

Reviewer 2: ```suggestion
| **Old Oxygen variable**         | **New Oxygen variable**     |
```

| ----------------------------------- | --------------------------- |
| SHOPIFY_STORE_DOMAIN                | PUBLIC_STORE_DOMAIN         |
| SHOPIFY_STOREFRONT_API_PUBLIC_TOKEN | PUBLIC_STOREFRONT_API_TOKEN |

> | SHOPIFY_STOREFRONT_API_SECRET_TOKEN | SECRET_STOREFRONT_API_TOKEN |

Reviewer: I advocated for `PRIVATE_`, to align with how the docs explain these variables (as public/private) to partners.

> | SHOPIFY_STOREFRONT_API_SECRET_TOKEN | SECRET_STOREFRONT_API_TOKEN |

Author: Let's roll with `PRIVATE_`, if that will be more consistent with the rest of Shopify

> | SHOPIFY_STOREFRONT_ID                    | PUBLIC_STOREFRONT_ID                 |

Reviewer 3: @graygilmore correct if I'm wrong but we are going to have to migrate some of the ENV vars on existing Hydrogen stores.

Here is a screenshot from my custom storefront production environment
<img width="1092" alt="image" src="https://user-images.githubusercontent.com/4672858/185144174-041a40b3-53b3-46f4-a7e8-321a3537f770.png">

Reviewer 3: @blittle I assume we are removing the `SHOPIFY_`  and `OXYGEN_` prefixes to make it more general? 

I wondering if `PUBLIC_STOREFRONT_ID` should be `PUBLIC_SHOPIFY_STOREFRONT_ID`. Since that `storefront_id` is always going to be associated with Shopify

Author: `PUBLIC_SHOPIFY_STOREFRONT_ID` makes sense to me
